crates/runtime/src/init/dataset.rs

```diff
@@ -202,6 +203,34 @@ impl Runtime {
             }
         };
 
+        // Register any component metrics that the user has enabled for this dataset.
+        if ds.metrics.has_enabled_metrics() {
```

Comment:
This is where we check what metrics the user has requested in the dataset definition and ask the data connector for its metrics provider to register them.